pytorch_tabnet/metrics.py

```diff
@@ -24,20 +24,26 @@ def UnsupervisedLoss(y_pred, embedded_x, obf_vars, eps=1e-9):
         Orginal input embedded by network
     obf_vars : torch.Tensor
         Binary mask for obfuscated variables.
-        1 means the variables was obfuscated so reconstruction is based on this.
+        1 means the variable was obfuscated so reconstruction is based on this.
     eps : float
         A small floating point to avoid ZeroDivisionError
         This can happen in degenerated case when a feature has only one value
 
     Returns
     -------
     loss : torch float
-        Unsupervised loss, average value over batch samples.    
+        Unsupervised loss, average value over batch samples.
     """
     errors = y_pred - embedded_x
-    reconstruction_errors = torch.mul(errors, obf_vars)**2
-    batch_stds = torch.std(embedded_x, dim=0)**2 + eps
+    reconstruction_errors = torch.mul(errors, obf_vars) ** 2
+    batch_stds = torch.std(embedded_x, dim=0) ** 2 + eps
     features_loss = torch.matmul(reconstruction_errors, 1 / batch_stds)
+    # compute the number of non-obfuscated variables
+    nb_used_variables = torch.sum(1 - obf_vars, dim=1)
```

Comment:
better have:
```python
    nb_masked_variables = torch.sum(masked_vars, dim=1)
```
than
```python
    nb_used_variables = torch.sum(1 - obf_vars, dim=1)
```

Reply:
well here we need the number of used variables and not the number of obfuscated variables so I'm not sure to understand your point?

Reply:
Indeed, that's what I mean, based on my understanding of the situation.
But maybe I am wrong. If so, a comment might explain somewhere why we need to 'correct' the `features_loss` when all variables are used (none hidden)

Reply:
Yes I think what we want to do is normalize the loss so that the range is not dependent on the number of reconstructed features. so we should indeed compute `nb_reconstructed_features = torch.sum(obf_vars, dim=1)`

Reply:
You're right, I mixed up the masks' meaning.